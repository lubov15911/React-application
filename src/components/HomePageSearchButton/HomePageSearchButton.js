import React from 'react';
import PropTypes from 'prop-types';

import './HomePageSearchButton.scss'

const HomePageSearchButton = (props) => {
    const { handleClick } = props;
    return (<button className="home-page-btn" onClick={handleClick} type="button">Search icon</button>);
};
HomePageSearchButton.propTypes = {
    handleClick: PropTypes.func.isRequired,
};

export default HomePageSearchButton;
