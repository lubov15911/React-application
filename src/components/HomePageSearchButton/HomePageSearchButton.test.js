import React from 'react';
import { shallow } from 'enzyme';

import HomePageSearchButton from './HomePageSearchButton';

describe('HomePageSearchButton', () => {
    it('should render correctly', () => {
        let routeToHomePage = () => {};
        const component = shallow(<HomePageSearchButton handleClick={routeToHomePage} />);

        expect(component).toMatchSnapshot();
    });
});
