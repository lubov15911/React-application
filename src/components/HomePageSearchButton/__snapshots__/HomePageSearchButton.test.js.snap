// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageSearchButton should render correctly with no props 1`] = `
<button
  className="home-page-btn"
  onClick={[Function]}
  type="button"
>
  Search icon
</button>
`;
